common/app/views/fragments/javaScriptFirstSteps.scala.html

```diff
@@ -194,35 +194,42 @@
 
 </script>
 
-
-<script>
-    guardian.isIphone6 = false;
-    guardian.isIphone4 = false;
-    guardian.inTestBucket = Math.floor(Math.random() * 2 + 1) === 1;
-</script>
-
 @if(IphoneConfidence.isSwitchedOn && Seq("uk", "us", "au").contains(item.id)) {
 
     @* Not the usual type of thing we do, but I want this separate to our "normal" javascript. *@
     @* also delete iPhoneConfidenceCheck inside of facia.js *@
 
-
     @* see http://www.paintcodeapp.com/news/ultimate-guide-to-iphone-resolutions *@
     <script>
+
+        function logDevice(model, os) {
+            (new Image()).src = guardian.config.page.beaconUrl + '/count/iphone-' + model + '-ios' + os + '-start.gif';
+
+            setTimeout(function() {
+                (new Image()).src = guardian.config.page.beaconUrl + '/count/iphone-' + model + '-ios' + os + '-after-5.gif';
+            }, 5000);
+
+            setTimeout(function() {
+                (new Image()).src = guardian.config.page.beaconUrl + '/count/iphone-' + model + '-ios' + os + '-after-10.gif';
+            }, 10000);
+        }
+
         if (navigator.platform === 'iPhone') {
 
-            guardian.isIphone6 = (screen.width === 375 && screen.height === 667) || (screen.width === 414 && screen.height === 736);
-            guardian.isIphone4 = screen.width === 320 && screen.height === 480 && window.devicePixelRatio > 1;
+            var iPhoneVersionRegex = /.*iPhone OS ([678])_\d+/;
 
-            if (guardian.isIphone6) {
-                (new Image()).src = guardian.config.page.beaconUrl + '/count/iphone-6-start.gif';
-            }
+            if (iPhoneVersionRegex.test(navigator.userAgent)) {
+
+                var osVersion = iPhoneVersionRegex.exec(navigator.userAgent)[1];
```

Comment:
It only gets that far if it passes line 221 `iPhoneVersionRegex.test(navigator.userAgent)` i.e. it matches the pattern.